src/util/amount.rs

```diff
@@ -1640,7 +1640,7 @@ mod tests {
 
     // Creates individual test functions to make it easier to find which check failed.
     macro_rules! check_format_non_negative {
-        ($denom:ident; $($test_name:ident, $val:expr, $format_string:expr, $expected:expr);* $(;)*) => {
+        ($denom:ident; $($test_name:ident, $val:expr, $format_string:expr, $expected:expr);* $(;)?) => {
```

Comment:
@Kixunil, one of the other task items in edition 2018 checklist mentions using `literal` instead of `expr`. This is the location here, right? I can do this change but I cannot justify it. I get that literal is more restrictive than expr, what is the benefit of being more restrictive? Or is there some other benefit? Thanks.

(FTR the justification for this PR is a bit vague as well.)

Reply:
If one accidentally passes non-literal the compiler should produce better errors because it realizes the argument is wrong at the place where the macro was used - the error wouldn't come out of inside of the macro.

This PR is purely cosmetic to avoid potentially confusing `some_macro!(a,b,c, ,,,,,,,);`

Reply:
re the literal/expr thing, cheers, that makes good sense. I'll patch it.

re the cosmetic thing, I guessed as much. Is there a specif reason that you didn't do this (put the semi-colon _inside_ the match pattern):
```
        ($denom:ident; $($test_name:ident, $val:expr, $format_string:expr, $expected:expr;)*) => {
```
Was it to be inline with how Rust allows for an optional comma? (Just out of interest and for my education.)

Reply:
`expr` -> `literal` change done in #1014

Reply:
Yes, trailing commas/semicolons are optional and putting it inside would make it mandatory.